staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -395,6 +395,16 @@ func chooseHostInterfaceFromRoute(routes []Route, nw networkInterfacer) (net.IP,
 				glog.V(4).Infof("Found active IP %v ", finalIP)
 				return finalIP, nil
 			}
+			glog.V(4).Infof("Default route exists for IPv%d, however interface %q does not have global unicast addresses. Checking loopback interface", uint(family), route.Interface)
```

Comment:
I'm not super familiar with this code, but won't this prefer a `lo` address before iterating through all of the provided routes?

It would naively seem to me that we want to iterate all the provided routes first, and if none of them provide an interface with a global unicast address we'd fallback on `lo`. Does that make sense?

Reply:
It will check if we have IPv4 default route, then will try to find unicast IPv4 address on route's interface or on 'lo'. If found valid unicast IPv4 address, it stops. Then the same thing will be done for IPv6.

The current code does practically the same: prefers IPv4 over IPv6 if default route for this family is found and we can determine interface address for this default route.